flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergStreamWriter.java

```diff
@@ -97,7 +98,12 @@ public String toString() {
         .toString();
   }
 
-  private void emit(WriteResult result) {
+  /** close all open files and emit files to downstream committer operator */
+  private void flush() throws IOException {
+    long startNano = System.nanoTime();
+    WriteResult result = writer.complete();
+    writerMetrics.updateFlushResult(result);
     output.collect(new StreamRecord<>(result));
+    writerMetrics.flushDuration(TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNano));
```

Comment:
Maybe it is clearer to just do use System.currentTimeMillis() and millis for the calculations since there is no extra precision from doing it in nanoseconds? Unless you want the extra precision, for which the return type long wouldn't suffice

Reply:
This is intentional (not for the reason of precision though). 

`nanoTime` is monotonically increasing number. it is best fit for measuring duration. `currentTimeMillis` can jump forward or backward in the case of clock adjustments (e.g. NTP).

https://stackoverflow.com/questions/2978598/will-system-currenttimemillis-always-return-a-value-previous-calls

Reply:
Thanks for the explanation!